core/sessions/authentication.go

```diff
@@ -0,0 +1,60 @@
+package sessions
+
+import (
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/v2/core/auth"
+	"github.com/smartcontractkit/chainlink/v2/core/bridges"
+)
+
+// Application config constant options
+type AuthenticationProviderName string
+
+const (
+	LocalAuth AuthenticationProviderName = "local"
+	LDAPAuth  AuthenticationProviderName = "ldap"
+)
+
+// ErrUserSessionExpired defines the error triggered when the user session has expired
+var ErrUserSessionExpired = errors.New("session missing or expired, please login again")
+
+// ErrNotSupported defines the error where interface functionality doesn't align with a Read Only LDAP server
+var ErrNotSupported = errors.New("functionality not supported with read only LDAP server")
```

Comment:
There is a new `errors.ErrUnsupported` that you could wrap here:
```suggestion
var ErrNotSupported = fmt.Errorf("functionality not supported with read only LDAP server: %w", errors.ErrUnsupported)
```
https://pkg.go.dev/errors#ErrUnsupported

Reply:
Oh interesting, do you think its better to keep this custom one defined for now?

Reply:
It is meant to be wrapped, so we can have both.